### PR TITLE
SDK-136 modified deploy module to work with server properties

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**
@rkorytkowski This is not finished yet. Any feedback will be appreciated :)

### CODE REVIEW

**Comment by AdamGrzybkowski** [comment]
@rkorytkowski 
I've implemented old functionality using properties from server, but now I have the following questions:
1) Should we keep war.openmrs... properties in server properties ? If yes what about openmrs.platform.version, delete it ? 
2) I've deleted userModules, because we keep current modules versions in server and distro properties. How openmrs-sdk:reset should work now ? 

I still have to add server validation for servers created with sdk version below 3.2.

**Comment by AdamGrzybkowski** [comment]
@rkorytkowski In the last commit I've added support for downgrading modules when deploying distro. I think if we support deleting we should also support downgrading. Openmrs Core can be downgraded to, but not deleted.

**Comment by AdamGrzybkowski** [comment]
@rkorytkowski I've resolved conflicts

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/model/ModuleProperties.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ *
+ */
+public class ModuleProperties {
```
Comment (PawelGutkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/model/ModuleProperties.java]: This name is a bit misleading, maybe we could name it "BaseSdkProperties". I guess this class could be abstract as well

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/model/Server.java [hunk 2]
```diff
@@ -658,11 +660,17 @@ public void setIncludeDemoData(boolean includeDemoData) {
...
 
-    public String getPlatformVersion(){
+    @Deprecated
+    public String getOpenmrsCoreVersion(){
         return getParam(PROPERTY_PLATFORM);
     }
-    public void setPlatformVersion(String platformVersion){
-        setParam(PROPERTY_PLATFORM, platformVersion);
+
+    public void saveOpenmrsCoreVersion(String version) throws MojoExecutionException {
```
Comment (PawelGutkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/model/Server.java]: Saving changes in distro properties file, but not changing openmrs-server properties file seems a bit incosistent and can possibly introduce bugs

**Pull request merged**